src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -1916,6 +1928,16 @@ WorkerExtendedOpNode(MultiExtendedOp *originalOpNode)
 	workerExtendedOpNode->limitCount = WorkerLimitCount(originalOpNode);
 	workerExtendedOpNode->sortClauseList = WorkerSortClauseList(originalOpNode);
 
+	/*
+	 * If grouped by a partition column whose values are distinct across shards,
+	 * we can push down the having qualifier.
+	 */
+	if (havingQual != NULL && groupedByDistinctPartitionColumn)
+	{
+		ereport(DEBUG1, (errmsg("pushing down HAVING")));
```

Comment:
I used ```DEBUG```s here because this file already used it for ```LIMIT``` push down. But I think we did that for ```LIMIT``` push-down because we didn't have distributed ```EXPLAIN``` when we first implemented it. Removed the ```DEBUG``` for ```HAVING``` and did as you suggested.

Reply:
Sorry about that feedback; I think before we had "true" `EXPLAIN` support, there was a Citus GUC to enable printing of the remote queries for the `Task`s. That's gone now, without a direct replacement, I guess.